src/v/cluster/archival/archival_metadata_stm.cc

```diff
@@ -885,11 +891,13 @@ ss::future<std::error_code> archival_metadata_stm::do_replicate_commands(
 
     // Create a promise to deliver the result of the batch application
     _active_operation_res.emplace();
-    auto broken_promise_to_shutdown = [](const ss::broken_promise&) {
-        return errc::shutting_down;
-    };
-    auto apply_result = _active_operation_res->get_future()
-                          .handle_exception_type(broken_promise_to_shutdown);
+
+    auto apply_result
+      = _active_operation_res->get_future()
+          .handle_exception_type(
+            treat_as_shutdown_exception<ss::broken_promise>)
+          .handle_exception_type(
+            treat_as_shutdown_exception<ssx::shutdown_requested_exception>);
     auto op_state_reset = ss::defer([&] { _active_operation_res.reset(); });
```

Comment:
i am wondering if it would be possible to use abort source that can be passed as an argument to `replicate_options` ?

Reply:
Not sure why, but abort source is not passed neither to _raft->replicate(), nor to wait_no_throw(). It can also be different in different calls. We can of course save a reference to it and trigger it, but who know what else is using it, maybe we break stopping order somewhere else?